File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int m) {
    vector<int> answer;
    int r = 0;
    int tmp_n = n;
    int tmp_m = m;
    
    while (tmp_m != 0){
        r = tmp_n % tmp_m;
        tmp_n = tmp_m;
        tmp_m = r;
    }
    
    answer.push_back(tmp_n);
    answer.push_back(n*m/tmp_n);
    
    return answer;
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {
    var answer = [];
    
    let bigger = -1;
    let smaller = -1;
    
    if (n >= m) {
        bigger = n;
        smaller = m;
    }
    else {
        bigger = m;
        smaller = n;
    }
    
    for (let i=smaller; i>=1; i--) {
        if (bigger % i == 0 && smaller % i == 0) {
            answer.push(i);
            break;
        }
    }
    
    let pivot = bigger;
    while (true) {
        if (pivot % smaller == 0 && pivot % bigger == 0) {
            answer.push(pivot);
            break;
        }
        pivot ++;
    }
    
    return answer;
}